Sources/Datadog/Logging/Log/LogBuilder.swift

```diff
@@ -27,7 +27,7 @@ internal struct LogBuilder {
 
     func createLogWith(level: LogLevel, message: String, date: Date, attributes: LogAttributes, tags: Set<String>) -> Log {
         return Log(
-            date: dateCorrector?.toServerDate(deviceDate: date) ?? date,
+            date: dateCorrector?.currentCorrection.adjustToServerDate(date) ?? date,
```

Comment:
`dateCorrector?.currentCorrection.applying(to: date)` sounds good to me, changed 👌.

On a wider note about `date.adjustingToServerDate()` - I don't like extending types coming from one module (here: `Foundation.Date`) with dependencies coming from another (here `Datadog.DateCorrection`). Good for verbosity in testing, but this introduces hard link between modules and is often annoying to decouple if things need to go modular. In most languages, where type extensions are not possible, this would even not compile, as it means circular dependency between modules.